Ednpoints.txt

```diff
@@ -0,0 +1,132 @@
+DELETE: /users/:userId - delete user's account
+
+GET: /dishes - get a list of dishes
+GET: /dishes/:dishId - get dish information
+GET: /dishes/popular - get popular dishes
+GET: /dishes/random - get some random dish
+
+GET: /chefs - get all chefs
+
+GET: /users/:userId/favourites?type=chef(dish) - get user's favourite dishes or chefs
+	or /users/:userId/favourite-dishes and /users/:userId/favourite-chefs
+POST, DELETE:
+/users/:userId/favourites?type=dishes - add/delete the dish to favourites (passing dishId)
+/users//:userId/favourites?type=chefs - add/delete the chef to favourites (passing chefId)
+
+**user's cart will be stored on the server instead of in local storage
+GET: /users/:userId/cart - get the cart info
+POST: /users/:userId/cart - add the dish to the cart (passing dishId)
+PATCH: /users/:userId/cart - update information about the dish in the cart (passing dishId)
+DELETE: /users/:userId/cart  - delete the dish from the cart (passing dishId)
+
+GET: /notifications/:userId - get notifications of this user
+
+POST: /reviews - create a new comment to this dish (pass dishId and userId)
+PATCH: /reviews/:reviewId - update user's comment to this dish
+DELETE: /reviews/:reviewId - delete user's comment to this dish
+
+GET: /orders/:userId - get user's orders list
+GET: /orders/:orderId - get a particular order
+POST: /orders - add a new order to user's orders list
+
+/************/
+
+CHEF ENDPOINTS:
+
+GET: /chefs/:chefId - get chef's info
+PATCH: /chefs/:chefId - update chef's info
+DELETE: /chefs/:chefId - delete chef's account
+
+GET: /dishes/:chefId - get all dishes of this chef
+POST: /dishes - create a new dish of this chef
+GET: /dishes/:dishId - get a particular dish
+PATCH: /dishes/:dishId - update a particular dish of this chef
+DELETE: /dishes/:dishId - delete a particular dish of this chef
+
+GET: /notifications/:chefId - get notifications of this chef
+
+GET: /orders/:chefId?status=(new, completed, canceled) - get chef's all orders
+GET: /orders/:orderId - get a particular order
+PATCH: /orders/:orderId - update status of this order
+
+POST: /reviews - create a new comment to this dish (passing dishId and chefId)
+PATCH: /reviews/:reviewId - update chef's comment to this dish
+DELETE: /reviews/:reviewId - delete chef's comment to this dish
+
+
+/************/
+
+DELIVERY PERSON ENDPOINTS:
+
+GET: /deliveries?status=(active) - get the list of all delivery persons, sorted by distance and availability based on delivery person's current location and other factors
+
+GET: /deliveries/:deliveryId - get delivery person information
+DELETE: /deliveries/:deliveryId - delete delivery person account
+PATCH: /deliveries/:deliveryId - update delivery profile info: toggle 'ready for work' status (active / not active), update 'current location' when active, update personal info
+
+GET: /orders/:deliveryId?status=(new, completed, canceled) - get info about all orders details according to their status
```

Comment:
do we need endpoint for accepting delivery request (to give courier opportunity to accept or cancel proposed order's delivery)?

Reply:
Definitely, there should be an endpoint here to change the status of the order PATCH: /orders/:orderId. Okay, we'll fix it